### PR TITLE
fixing show interfaces summary for 10/40/100 Gig Interfaces

### LINKED ISSUE #1154: Cisco IOS Interface Counter TenGig and beyond not parsed correctly
### napalm version
(Paste verbatim output from `pip freeze | grep napalm` between quotes below)

```
napalm            2.5.0
****
```

### Network Operating Sytem 
```
Cisco IOS XE Software, Version 16.05.01b
```

I was playing arround with napalm to gather some interface data specifically on ios xe  router with 10G interfaces.

I noticed that I had a couple of interfaces with drops, but using the function 
get_interface_counters through nornir delivers a tx_discards: -1 

```
  'TenGigabitEthernet0/0/2': { 'rx_broadcast_packets': 0,
                               'rx_discards': -1,
                               'rx_errors': 0,
                               'rx_multicast_packets': 0,
                               'rx_octets': 1533128411516,
                               'rx_unicast_packets': 1780831170,
                               'tx_broadcast_packets': -1,
                               'tx_discards': -1,
                               'tx_errors': 0,
                               'tx_multicast_packets': -1,
                               'tx_octets': 1006199251542,
                               'tx_unicast_packets': 1631763809},

```

I could not image how -1 could be the correct output. 

Interfaces with Gigabitspeed were parsed correctly. 

IOS.py Line 2105

Digging through the source code I found , that napalm using "show interface summary" to get the IQD value, but it is searching for TenGigabitEthernet0/0/2 not for Te0/0/2 like in the output. 

###show interface summary example 
```
   Interface                   IHQ       IQD       OHQ       OQD      RXBS      RXPS      TXBS      TXPS      TRTL    
 -----------------------------------------------------------------------------------------------------------------    
 * Te0/0/0                       0         0         0         0 633161000    222861 1567988000    276185         0   
 * Te0/0/1                       0         0         0         0 514257000    153657 2048510000    343953         0   
 * Te0/0/2                       0         0         0         45454 514257000    153657 2048510000    343953         0    
 * GigabitEthernet0/0            0         0         0         1         0         1         0         0         0    
   GigabitEthernet0/1            0         0         0         2         0         1         0         0         0    
   GigabitEthernet0/2            0         0         0         0         0         1         0         0         0    
```                                                                                                                

From my point of view FortyGigabitEthernet , TwentyFiveGigabitEthernet ,HundredGigabitEthernet
are also not parsed correctly. 

I have already created a workaround, but not sure if this fits the code qualtity of this project. 

###Prototype
```
 for line in text.splitlines():
	 short_if_names = {'HundredGigabitEthernet':'Hu','TwentyFiveGigabitEthernet':'Tf','FortyGigabitEthernet':'Fo','TenGigabitEthernet':'Te'}
     match = re.search(r'(HundredGigabitEthernet|TwentyFiveGigabitEthernet|FortyGigabitEthernet|TenGigabitEthernet)(\d\/\d\/\d)',
             interface)

     if match:

        interface_short = '{}{}'.format(short_if_names.get(\
        match.groups()[0]), match.groups()[1])
        if interface_short in line:
             # Line is tabular output with columns
             # Interface  IHQ  IQD  OHQ  OQD  RXBS  RXPS  TXBS  TXPS  TRTL
             # where columns (excluding interface) are integers
             regex = (
                 r"\b"
                 + interface_short
                 + r"\b\s+(\d+)\s+(?P<IQD>\d+)\s+(\d+)"
                 + r"\s+(?P<OQD>\d+)\s+(\d+)\s+(\d+)"
                 + r"\s+(\d+)\s+(\d+)\s+(\d+)"
             )
             match = re.search(regex, line)
             if match:
				counters[interface]["rx_discards"] = int(match.group("IQD"))
				counters[interface]["tx_discards"] = int(match.group("OQD"))
     else:

         
         if interface in line:
             # Line is tabular output with columns
             # Interface  IHQ  IQD  OHQ  OQD  RXBS  RXPS  TXBS  TXPS  TRTL
             # where columns (excluding interface) are integers
             regex = (
                 r"\b"
                 + interface
                 + r"\b\s+(\d+)\s+(?P<IQD>\d+)\s+(\d+)"
                 + r"\s+(?P<OQD>\d+)\s+(\d+)\s+(\d+)"
                 + r"\s+(\d+)\s+(\d+)\s+(\d+)"
             )
             match = re.search(regex, line)
             if match:
				counters[interface]["rx_discards"] = int(match.group("IQD"))
				counters[interface]["tx_discards"] = int(match.group("OQD"))
```

### PR DESCRIPTION
**Pull request by network-shark**
This PR fixes #1154

### CODE REVIEW

**Comment by network-shark** [comment]
@mirceaulinic I have uploaded the PR again , but I don't know how to fix the black test.... .

File: napalm/1 [hunk 1]
```diff
@@ -0,0 +1,3501 @@
+"""NAPALM Cisco IOS Handler."""
```
Comment (mirceaulinic) [inline napalm/1:1]: Can you remove this?

File: napalm/ios/ios.py [hunk 2]
```diff
@@ -2102,6 +2104,12 @@ def get_interfaces_counters(self):
                     match = re.search(r"(\d+) output errors", line)
                     counters[interface]["tx_errors"] = int(match.group(1))
                     counters[interface]["tx_discards"] = -1
+
+            interface_type, interface_number = split_interface(interface)
+            if interface_type in[
```
Comment (mirceaulinic) [inline napalm/ios/ios.py:2109]: See https://github.com/napalm-automation/napalm/pull/1161/commits/2705cb17f849a29e0254caaf7f0ec06e52610834

**Comment by network-shark** [comment]
Should I add  the line to 
test/ios/mocked_data/test_get_interfaces_counters/normal/show_interface_summary.txt ?

```
* Te0/0/0                       0         0         0         0 633161000    222861 1567988000    276185         0
```

File: napalm/1 (hunk 1 above)
Comment (mirceaulinic) [reply napalm/1:1]: I meant to remove this file.

File: napalm/ios/ios.py [hunk 3]
```diff
@@ -1,4 +1,3 @@
-"""NAPALM Cisco IOS Handler."""
```
Comment (mirceaulinic) [inline napalm/ios/ios.py:1]: Please don't remove this line.

**Comment by mirceaulinic** [comment]
> Should I add some test data to
test/ios/mocked_data/test_get_interfaces_counters/normal/show_interface_summary.txt ?

I'd suggest adding a separate test case, e.g., under `test/ios/mocked_data/test_get_interfaces_counters/<test case name>/show_interface_summary.txt` (you'll also need to provide `test/ios/mocked_data/test_get_interfaces_counters/<test case name>/expected_result.json`. Cheers.

File: napalm/1 (hunk 1 above)
Comment (mirceaulinic) [inline napalm/1:1]: @network-shark you still didn't remove this file.

File: test/ios/mocked_data/test_get_interfaces_counters/short_if_names/expected_result.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{
+	'TenGigabitEthernet0/0/0': {
```
Comment (mirceaulinic) [inline test/ios/mocked_data/test_get_interfaces_counters/short_if_names/expected_result.json:2]: That's not valid JSON.

File: napalm/ios/ios.py [hunk 5]
```diff
@@ -2103,6 +2105,14 @@ def get_interfaces_counters(self):
                     match = re.search(r"(\d+) output errors", line)
                     counters[interface]["tx_errors"] = int(match.group(1))
                     counters[interface]["tx_discards"] = -1
+
+            interface_type, interface_number = split_interface(interface)
+            if interface_type in [
+                'HundredGigabitEthernet',
```
Comment (mirceaulinic) [inline napalm/ios/ios.py:2111]: Okay, only Black is failing now (seems like it requires double quotes on these): just run `black .` (in the environment you installed `requirements-dev.txt`) and should be gtg.

Comment (network-shark) [reply napalm/ios/ios.py]: Thank you for beeing such a help :)

**Pull request merged**